docs/src/prototypes/mentions/MentionsDropdown.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import { Dropdown, DropdownItemProps, Provider } from '@stardust-ui/react'
+
+import { MentionsContainerProps } from './MentionsEditor'
+
+const MentionsDropdown: React.FunctionComponent<MentionsContainerProps> = props => {
+  const { searchQuery, items, onOpenChange, onSearchQueryChange, onInputKeyDown } = props
+
+  const renderItem = React.useCallback(
+    (Item: React.ReactType, props: DropdownItemProps, boldFont: number) => {
+      const { header = '', ...rest } = props
+      const headerStr = String(header)
+      const queryStartIndex = headerStr.indexOf(searchQuery)
+
+      if (queryStartIndex < 0) return <Item {...props} />
+      const queryEndIndex = queryStartIndex + searchQuery.length
+
+      return (
+        <Item
+          {...rest}
+          header={{
+            content: (
+              <span>
+                {headerStr.substring(0, queryStartIndex)}
+                <span style={{ fontWeight: boldFont }}>
+                  {headerStr.substring(queryStartIndex, queryEndIndex)}
+                </span>
+                {headerStr.substring(queryEndIndex)}
+              </span>
```

Comment:
```jsx
const DropdownItemFilteredHeader = (props) => {
  const { header, queryStartIndex } = props

  // logic

  return condition ? <ListItemHeader {...} /> : <span>{...}</span>
}
```

Why we can't create a custom component there instead of `renderMethod()`?

Reply:
@layershifter - plenty of reasons:

- more difficult to customize because we need to create new component with dependency on `searchQuery` and styles (`fontWeight`)

```jsx
const DropdownItemFilteredHeader = (
  props: DropdownItemProps & { searchQuery?: string; fontWeight?: FontWeightProperty },
) => {
  const { fontWeight, searchQuery, header, ...rest } = props

  // logic
  
  return (
    <DropdownItem
      {...rest}
      header={{
        content: (
          <span>
            {header.substring(0, queryStartIndex)}
            <span style={{ fontWeight }}>{header.substring(queryStartIndex, queryEndIndex)}</span>
            {header.substring(queryEndIndex)}
          </span>
        ),
      }}
    />
  )
}
```

- we cannot reuse the prop overrides for Dropdown items (`onRemove`, `onClick`, `onKeydown`) => broken functionality

- broken accessibility

- need to define `key` prop in items:

```jsx
items={items.map((item, index) => (
  <DropdownItemFilteredHeader key={index} {...item} />
))}
```

Reply:
If we will use `items.map(item => { ...item, as: DropdownItemFilteredHeader })` everything should work.

But it will break custom `as`...

Reply:
@layershifter - can we agree to leave as is then? I still don't see much benefit to using map vs `renderItem`